plugins-extra/query_enhancements/public/plugin.ts

```diff
@@ -31,6 +31,14 @@ export class QueryEnhancementsPlugin
       usageCollector: data.search.usageCollector,
     });
 
+    const sqlAsyncSearchInterceptor = new SQLAsyncQlSearchInterceptor({
+      toasts: core.notifications.toasts,
+      http: core.http,
+      uiSettings: core.uiSettings,
+      startServices: core.getStartServices(),
+      usageCollector: data.search.usageCollector,
```

Comment:
we should use the usagecollector to see if they optin into using this async queries. ask me if you ened more details